File: GarbageV3/app/src/main/java/dk/itu/garbagev3/MainActivity.java
package dk.itu.garbagev3;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    private EditText inputGarbage;
    private ItemsDB itemsDB;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        inputGarbage = findViewById(R.id.input_garbage);
        Button Where = findViewById(R.id.where_button);

        itemsDB= ItemsDB.get();
        itemsDB.fillItemsDB(MainActivity.this,"garbage.txt");


        Where.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                String itemName = inputGarbage.getText().toString().trim();
                if (itemName.isEmpty()) {
                    Toast.makeText(MainActivity.this, "Please enter an item!", Toast.LENGTH_SHORT).show();
                } else{
                    String category = itemsDB.findCategory(itemName);
                    inputGarbage.setText(category);
                }
            }
        });

        Button add= findViewById(R.id.add_a_button);
        add.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent= new Intent(MainActivity.this, AddActivity.class);
                startActivity(intent);
            }
        });
    }
}

File: KotlinGarbageV4/app/src/main/java/dk/itu/kotlingarbagev4/MainActivity.kt
package dk.itu.kotlingarbagev4

import androidx.appcompat.app.AppCompatActivity

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import android.widget.TextView
import android.widget.Toast



class MainActivity : AppCompatActivity(){
    private lateinit var itemsDB: ItemsDB

    override fun onCreate(savedInstanceState: Bundle?){
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val inputGarbage:TextView = findViewById(R.id.input_garbage)
        itemsDB = ItemsDB.get()
        itemsDB.fillItemsDB(this, "garbage.txt")

        val where: Button = findViewById(R.id.where_button)
        where.setOnClickListener{
            val itemName = inputGarbage.text.toString().trim()
            if(itemName.isEmpty()) {
                Toast.makeText(this, "Please enter an item!", Toast.LENGTH_SHORT).show()
            } else{
                val category:String = itemsDB.findCategory(itemName)
                inputGarbage.text = category
            }
        }

        val add: Button = findViewById(R.id.add_a_button)
        add.setOnClickListener {
            val intent = Intent(this, AddActivity::class.java)
            startActivity(intent)
        }
    }

    //empty the input area
    override fun onResume() {
        super.onResume()
        val inputGarbage:TextView = findViewById(R.id.input_garbage)
        inputGarbage.text=""
    }
}
